test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/ExpressionMetadataProviderTest.cs

```diff
@@ -148,6 +148,29 @@ public void FromStringExpression_SetsContainerAsExpected()
             Assert.Same(myModel, metadata.Container.Model);
         }
 
+        // This is a special case that's basically only possible with Razor Pages.
```

Comment:
Aim here is to make at least the cases where `metadataProvider.GetMetadataForType(containerType).Properties[propertyName]` is called but can fail behave consistently. Secondary concern is providing reasonable error messages for the cases that will fail just after this method returns.